orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+                return tokens[doc][index]
+            if col == 2:
+                if index + self.width < len(tokens[doc]):
+                    return ' '.join(tokens[doc][index + 1:index + self.width])
+                else:
+                    return ' '.join(tokens[doc][index + 1:])
+
+        elif role == Qt.TextAlignmentRole:
+            return [Qt.AlignRight | Qt.AlignVCenter, Qt.AlignCenter, Qt.AlignLeft | Qt.AlignVCenter][col]
+
+        elif role == Qt.ForegroundRole:
+            if col == 1:
+                return QBrush(QColor(0, 0, 255))
+
+        elif role == LastDocumentRole:
+             return row < len(self.word_index) - 1 and doc < self.word_index[row + 1][0]
+
+    def _compute_indices(self):  # type: () -> Optional[None, list]
+        if self.data is None:
+            self.indices = None
+            return
+        if self.data and not self.data.has_tokens():
+            preprocessor = Preprocessor(tokenizer=WordPunctTokenizer())
+            preprocessor(self.data)
+        self.indices = [ConcordanceIndex(doc, key=lambda x: x.lower()) for doc in self.data.tokens]
+
+    def matching_docs(self):
+        if self.indices and self.word:
+            return sum(bool(doc.offsets(self.word)) for doc in self.indices)
+        else:
+            return 0
+
+
+class OWConcordance(OWWidget):
+    name = "Concordance"
+    description = "Display the context of the word."
+    icon = "icons/Concordance.svg"
+    priority = 30000
+
+    inputs = [('Corpus', Table, 'set_data')]
+    outputs = [('Selected Documents', Table, )]
+
+    autocommit = Setting(True)
+    width = Setting(5)
+    word = Setting("")
+    # TODO Set selection settings.
+
+    def __init__(self):
+        super().__init__()
+
+        self.corpus = None      # Corpus
+        self.n_documents = ''   # Info on docs
+        self.n_matching = ''    # Info on docs matching the word
+        self.n_tokens = ''      # Info on tokens
+        self.n_types = ''       # Info on types (unique tokens)
+
+        # Info attributes
+        info_box = gui.widgetBox(self.controlArea, 'Info')
+        gui.label(info_box, self, 'Documents: %(n_documents)s')
+        gui.label(info_box, self, 'Tokens: %(n_tokens)s')
+        gui.label(info_box, self, 'Types: %(n_types)s')
+        gui.label(info_box, self, 'Matching: %(n_matching)s')
+
+        # Width parameter
+        gui.spin(self.controlArea, self, 'width', 3, 10, box=True, label="Number of words:", callback=self.set_width)
+
+        gui.rubber(self.controlArea)
+
+        # Search
+        c_box = gui.widgetBox(self.mainArea, orientation="vertical")
+        self.input = gui.lineEdit(c_box, self, 'word',
+                                         orientation=Qt.Horizontal,
+                                         sizePolicy=QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed),
+                                         label='Query:', callback=self.refresh_search, callbackOnType=True)
```

Comment:
Does this work better:

```
self.input = gui.lineEdit(
            c_box, self, 'word', orientation=Qt.Horizontal,
            sizePolicy=QSizePolicy(QSizePolicy.MinimumExpanding, 
                                   QSizePolicy.Fixed), 
            label='Query:', callback=self.refresh_search, callbackOnType=True)
```
QSizePolicy goes over the 80 characters, so I had to break it.